### DISCUSSION TITLE
v9.0.0 (2023-10-22)

### DISCUSSION THREAD

**Discussion by yajo** [copier-org/copier#1377] [General]
### BREAKING CHANGE

- Changes the return code for unsafe template error from
2 to 4 to avoid return code collision with Plumbum's `SwitchError` type
errors which use return code 2.

### Feat

- allow overriding data file with CLI arguments (#1332)
- **cli**: read answers from yaml file (#1325)

### Fix

- **vcs**: prevent local clone from being stuck by gpg prompts (#1360)
- **pydantic**: compatible with 2.4+, solving `FieldValidationInfo` warning (#1342)
- fix answer validation for `type: str` questions
- version guess from tags that don't start with `v`, but are still PEP440 compliant (#1048)
- **cli**: use return code 4 for unsafe template error
- **nix**: detect build version appropriately

<hr /><em>This discussion was created from the release <a href='https://github.com/copier-org/copier/releases/tag/v9.0.0'>v9.0.0 (2023-10-22)</a>.</em>

**Comment by sisp** [comment]
@yajo I just realized that the PyPI package for v9.0.0 is missing: https://pypi.org/project/copier/#history

**Reply by yajo** [reply]
Ah yeah, because of the ci failure. I'll publish a new release soon.

**Reply by yajo** [reply]
Done. I hope this new one gets to pypi!

**Reply by sisp** [reply]
v9.0.0 is still not on pypi.org.

**Reply by pawamoy** [reply]
But 9.0.1 is. It should be possible for @yajo to manually push 9.0.0 from his workstation, without passing through CI. Something like `git checkout 9.0.0; poetry build; poetry publish`.

**Reply by pawamoy** [reply]
By the way there a few warnings coming from the use of `pypa/gh-action-pypi-publish@master` instead of `pypa/gh-action-pypi-publish@release/v1` (see annotations in https://github.com/copier-org/copier/actions/runs/6690863211).
